### PR TITLE
[METAED-1374] rewrite that removes invasive behaviors

### PR DESCRIPTION
**Pull request by bradbanister**

### CODE REVIEW

**Comment by bradbanister** [comment]
Incorporated changes from comments in other PR. Sets ruler to 80 just for markdown files.

File: .vscode/settings.json [hunk 1]
```diff
@@ -7,7 +7,11 @@
     { "language": "javascript", "autoFix": true },
     { "language": "typescript", "autoFix": true }
   ],
-  "editor.rulers": [80, 125],
+  "editor.rulers": [125],
+  "[markdown]": {
+    "editor.rulers": [80]
+  },
```
Comment (stephenfuqua) [inline .vscode/settings.json:13]: 😄

File: package.json [hunk 2]
```diff
@@ -147,20 +137,9 @@
 						"I do not consent."
 					]
 				},
-				"metaed.dataStandardProjectDirectory": {
-					"order": 6,
-					"description": "Full path to the set of core MetaEd files for the Ed-Fi Data Standard.",
-					"type": "string",
-					"default": ""
-				},
```
Comment (stephenfuqua) [inline package.json:155]: With this setting removed, how do we select which data standard to use?

File: package.json [hunk 3]
```diff
@@ -79,10 +74,6 @@
 					"command": "metaed.build",
 					"group": "navigation"
 				},
-				{
-					"command": "metaed.about",
-					"group": "navigation"
-				},
```
Comment (stephenfuqua) [inline package.json:85]: Why remove the About button?

File: package.json (hunk 2 above)
Comment (stephenfuqua) [reply package.json:155]: I went back to the readme and now realize that it is suggesting that I need to load the Data Standard folder manually in the Explorer.

File: src/client/ServerMessageFactory.ts [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+      outputChannel,
+      showUiNotifications,
+    );
+    return undefined;
+  }
+
+  if (invalidProjects.length !== 0) {
+    // There are non-MetaEd projects in the workspace
+    await notifyInfo(
+      'There are non-MetaEd projects in the workspace. They will be ignored',
```
Comment (stephenfuqua) [inline src/client/ServerMessageFactory.ts:65]: I'm getting this message on my extension folder, and it is preventing me from running any actions on the extension.
image

File: src/client/ProjectFinder.ts [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+  // eslint-disable-next-line no-restricted-syntax
+  for (const folderPath of folderPaths) {
+    const projectJsonFilePath = path.join(folderPath, PROJECT_SETTINGS_FILE_NAME);
+    const projectJsonMetadata: ProjectJsonFields | null = await projectMetadataFromProjectJson(projectJsonFilePath);
+
+    if (projectJsonMetadata == null) {
+      invalidProjects.push({
+        folderPath,
+        reasonInvalid:
+          'Workspace folder have a package.json file with both metaEdProject.projectName and metaEdProject.projectVersion definitions.',
```
Comment (stephenfuqua) [inline src/client/ProjectFinder.ts]: Is this missing "does not"?
```suggestion
          'Workspace folder does not have a package.json file with both metaEdProject.projectName and metaEdProject.projectVersion definitions.',
```

File: package.json (hunk 2 above)
Comment (bradbanister) [reply package.json:155]: You just add it to your workspace. See the readme on how to find them.

File: package.json (hunk 3 above)
Comment (bradbanister) [reply package.json:85]: The "About" function is handled by the readme.md being the extension "home" page. The only other thing it did was listed the installed plugins, but you can get that from the Output window.

File: src/client/ServerMessageFactory.ts (hunk 4 above)
Comment (bradbanister) [reply src/client/ServerMessageFactory.ts:65]: Capital N?

File: src/client/ProjectFinder.ts (hunk 5 above)
Comment (bradbanister) [reply src/client/ProjectFinder.ts]: 😄

File: src/client/ServerMessageFactory.ts (hunk 4 above)
Comment (stephenfuqua) [reply src/client/ServerMessageFactory.ts:65]: The blue squiggle was from the spellchecker, and I ignored that. But you're right, that was the problem. Things are working much better now :-).

Comment (stephenfuqua) [reply src/client/ServerMessageFactory.ts:65]: This is a good one for me to include in the troubleshooting guide.

**Review by stephenfuqua — APPROVED** [review]

**Pull request merged**